server/api/sample_sizes.py

```diff
@@ -1,14 +1,72 @@
-from typing import Dict
+from typing import Dict, Optional
 from collections import Counter
+from itertools import product
 from flask import jsonify
-from werkzeug.exceptions import BadRequest, Conflict
+from werkzeug.exceptions import BadRequest
 
 from . import api
 from ..models import *  # pylint: disable=wildcard-import
 from ..auth import restrict_access, UserType
-from ..audit_math import ballot_polling, macro, supersimple, sampler_contest
+from ..audit_math import (
+    ballot_polling,
+    macro,
+    supersimple,
+    sampler_contest,
+    suite,
+)
 from . import rounds  # pylint: disable=cyclic-import
-from .cvrs import all_cvrs_uploaded
+from .cvrs import validate_uploaded_cvrs, hybrid_contest_choice_vote_counts
+from .ballot_manifest import validate_uploaded_manifests, hybrid_contest_total_ballots
+
+
+def suite_misstatements(
+    contest: sampler_contest.Contest,
+    reported_results: sampler_contest.CVRS,
+    audited_results: sampler_contest.SAMPLECVRS,
+) -> suite.MISSTATEMENTS:
+    def misstatement(
+        winner: str,
+        loser: str,
+        reported: Optional[sampler_contest.CVR],
+        audited: Optional[sampler_contest.CVR],
+    ) -> int:
+        # Special cases: if ballot wasn't in CVR or ballot can't be found by
+        # audit board, count it as a two-vote overstatement
+        if reported is None or audited is None:
+            return 2
+
+        v_w, v_l = (
+            (reported[contest.name][winner], reported[contest.name][loser])
+            if contest.name in reported
+            # If contest wasn't on the ballot according to the CVR
+            else (0, 0)
+        )
+
+        a_w, a_l = (
+            (audited[contest.name][winner], audited[contest.name][loser])
+            if contest.name in audited
+            # If contest wasn't on the ballot according to the audit board
+            else (0, 0)
+        )
+
+        return (v_w - a_w) - (v_l - a_l)
+
+    misstatements: suite.MISSTATEMENTS = {}
+    for winner, loser in product(contest.winners, contest.losers):
+        misstatement_counts = Counter(
+            misstatement(
+                winner, loser, audited_result["cvr"], reported_results.get(ballot)
+            )
+            for ballot, audited_result in audited_results.items()
+        )
+        misstatements[(winner, loser)] = {
+            "o1": misstatement_counts[1],
```

Comment:
whoa this is a really cool way to do this

Reply:
yeah, discovered this when we did ballot comparison!